pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -122,3 +132,90 @@ func buildMixerOutboundTCPFilter(env *model.Environment, node *model.Proxy) list
+// (forward attributes, but not invoke check calls)  ServiceInstances belong to the Node.
+func buildHTTPMixerFilterConfig(mesh *meshconfig.MeshConfig, role model.Proxy, nodeInstances []*model.ServiceInstance, outboundRoute bool, config model.IstioConfigStore) *mccpb.HttpClientConfig { // nolint: lll
+	transport := &mccpb.TransportConfig{
+		CheckCluster:  v1.MixerCheckClusterName,
+		ReportCluster: v1.MixerReportClusterName,
+	}
+
+	mxConfig := &mccpb.HttpClientConfig{
+		MixerAttributes: &mpb.Attributes{
+			Attributes: map[string]*mpb.Attributes_AttributeValue{},
+		},
+		ServiceConfigs: map[string]*mccpb.ServiceConfig{},
+		Transport:      transport,
+	}
+
+	var labels map[string]string
+	// Note: instances are all running on mode.Node named 'role'
+	// So instance labels are the workload / Node labels.
+	if len(nodeInstances) > 0 {
+		labels = nodeInstances[0].Labels
+		mxConfig.DefaultDestinationService = nodeInstances[0].Service.Hostname
+	}
+
+	if !outboundRoute {
+		// for outboundRoutes there are no default MixerAttributes
+		// specific MixerAttributes are in per route configuration.
+		v1.AddStandardNodeAttributes(mxConfig.MixerAttributes.Attributes, v1.AttrDestinationPrefix, role.IPAddress, role.ID, labels)
+	}
+
+	if role.Type == model.Sidecar && !outboundRoute {
+		// Don't forward mixer attributes to the app from inbound sidecar routes
+	} else {
+		mxConfig.ForwardAttributes = &mpb.Attributes{
+			Attributes: map[string]*mpb.Attributes_AttributeValue{},
+		}
+		addStandardNodeAttributes(mxConfig.ForwardAttributes.Attributes, v1.AttrSourcePrefix, role.IPAddress, role.ID, labels)
+	}
+
+	for _, instance := range nodeInstances {
+		mxConfig.ServiceConfigs[instance.Service.Hostname] = v1.ServiceConfig(instance.Service.Hostname, instance, config,
+			outboundRoute || mesh.DisablePolicyChecks, outboundRoute)
+	}
+
+	return mxConfig
+}
+
+// buildTCPMixerFilterConfig builds a TCP filter config for inbound requests.
+func buildTCPMixerFilterConfig(mesh *meshconfig.MeshConfig, role model.Proxy, instance *model.ServiceInstance) *mccpb.TcpClientConfig {
+	attrs := v1.StandardNodeAttributes(v1.AttrDestinationPrefix, role.IPAddress, role.ID, nil)
+	attrs[v1.AttrDestinationService] = &mpb.Attributes_AttributeValue{Value: &mpb.Attributes_AttributeValue_StringValue{instance.Service.Hostname}}
+
+	mxConfig := &mccpb.TcpClientConfig{
+		MixerAttributes: &mpb.Attributes{
+			Attributes: attrs,
+		},
+		Transport: &mccpb.TransportConfig{
+			CheckCluster:  v1.MixerCheckClusterName,
+			ReportCluster: v1.MixerReportClusterName,
+		},
+		DisableCheckCalls: mesh.DisablePolicyChecks,
+	}
+
+	return mxConfig
+}
+
+// addStandardNodeAttributes add standard node attributes with the given prefix
+func addStandardNodeAttributes(attr map[string]*mpb.Attributes_AttributeValue, prefix string, IPAddress string, ID string, labels map[string]string) {
```

Comment:
is this function used?   Is it the same as v1.AddStandardNodeAttributes?

Reply:
it's the same. that was exactly my point earlier in the review where @rshriram requested this to be copied in. imo there's no point copying if the code is identical.